source/event_stream_rpc_client.c

```diff
@@ -0,0 +1,979 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/event-stream/event_stream_channel_handler.h>
+#include <aws/event-stream/event_stream_rpc_client.h>
+#include <aws/event-stream/private/event_stream_rpc_priv.h>
+
+#include <aws/common/atomics.h>
+#include <aws/common/hash_table.h>
+#include <aws/common/mutex.h>
+
+#include <aws/io/channel_bootstrap.h>
+
+#include <inttypes.h>
+
+#ifdef _MSC_VER
+/* allow declared initializer using address of automatic variable */
+#    pragma warning(disable : 4221)
+/* allow non-constant aggregate initializers */
+#    pragma warning(disable : 4204)
+
+#endif
+
+struct aws_event_stream_rpc_client_connection {
+    struct aws_allocator *allocator;
+    struct aws_hash_table continuation_table;
+    struct aws_client_bootstrap *bootstrap_ref;
+    struct aws_atomic_var ref_count;
+    struct aws_channel *channel;
+    struct aws_channel_handler *event_stream_handler;
+    uint32_t latest_stream_id;
+    struct aws_mutex stream_lock;
+    struct aws_atomic_var is_closed;
+    struct aws_atomic_var handshake_complete;
+    size_t initial_window_size;
+    aws_event_stream_rpc_client_on_connection_setup_fn *on_connection_setup;
+    aws_event_stream_rpc_client_connection_protocol_message_fn *on_connection_protocol_message;
+    aws_event_stream_rpc_client_on_connection_shutdown_fn *on_connection_shutdown;
+    void *user_data;
+    bool bootstrap_owned;
```

Comment:
is this `owned_by_bootstrap` or `owns_bootstrap`, or `someone_else_owns_bootstrap`?

Reply:
bootstrap owns the connection. This is largely just a hack to allow testing channel where bootstrap callbacks aren't ever invoked.